lib/ingram_micro/elements/sales_order_line_item.rb

```diff
@@ -1,37 +1,79 @@
-class IngramMicro::SalesOrderLineItem < IngramMicro::BaseElement
+module IngramMicro
+  class SalesOrderLineItem < BaseElement
+    DEFAULTS = {
+      line_no: nil,
+      item_code: nil,
+      universal_product_code: nil,
+      product_name: nil,
+      comments: nil,
+      quantity: 1.0,
+      unit_of_measure: nil,
+      sid: nil,
+      esn: nil,
+      min: nil,
+      mdn: nil,
+      irdb: nil,
+      imei: nil,
+      market_id: nil,
+      line_status: nil,
+      base_price: 0.0,
+      line_discount: 0.0,
+      line_tax1: 0.0,
+      line_tax2: 0.0,
+      line_tax3: 0.0,
+      special_message: nil,
+      line_name_value: nil
+    }.freeze
 
-  DEFAULTS = {
-    :line_no => nil,
-    :item_code => nil,
-    :universal_product_code => nil,
-    :product_name => nil,
-    :comments => nil,
-    :quantity => 1.0,
-    :unit_of_measure => nil,
-    :sid => nil,
-    :esn => nil,
-    :min => nil,
-    :mdn => nil,
-    :irdb => nil,
-    :imei => nil,
-    :market_id => nil,
-    :line_status => nil,
-    :base_price => 0.0,
-    :line_discount => 0.0,
-    :line_tax1 => 0.0,
-    :line_tax2 => 0.0,
-    :line_tax3 => 0.0
-  }
+    attr_reader :line_name_value
 
-  def line_no
-    @element[:line_no]
-  end
+    def defaults
+      DEFAULTS
+    end
 
-  def line_no=(num)
-    @element[:line_no] = num
-  end
+    def line_no
+      @element[:line_no]
+    end
+
+    def line_no=(num)
+      @element[:line_no] = num
+    end
+
+    def build(builder)
+      # Similar to BaseElement except we want to skip special_message and
+      # line_name_value and handle them differently.
+      defaults.keys.each do |field|
+        next if field == (:special_message || :line_name_value)
+        element_name = field.to_s.tr('_', '-')
+        element_value = formatted_value_of(field)
+        builder.send(element_name, element_value)
+      end
+      add_special_message(builder)
+      # Expect that line_name_value will be an array of arrays, each of which
+      # contains a name and value that will translate to line-attribute-name and
+      # line-attribute-value.
+      unless element[:line_name_value].nil?
+        element[:line_name_value].each do |pair|
+          attr_name, attr_value = pair
```

Comment:
It's being assigned to the second element in pair, which is an array. Perhaps this isn't obvious and I should instead do `attr_name = pair[0]` and `attr_value = pair[1]`